### DISCUSSION TITLE
How to capture tls_handshake errors?

### DISCUSSION THREAD

**Discussion by changsin** [mitmproxy/mitmproxy#6584] [General]
How can I capture any tls_handshake errors? I tried writing a simple script like the following but it's not capturing the event.

```
def tls_handshake(flow):
    if flow.error:
        print(f"TLS Handshake failed for {flow.server_conn.address}")
        print(f"Error: {flow.error}")

addons = [
    tls_handshake
]
```

TLS handshake errors are displayed by mitmdump but not mitmproxy and it makes error handling and debugging difficult.

**Comment by mhils** [comment]
Take a look at https://docs.mitmproxy.org/stable/api/events.html#TLSEvents
:)

**Reply by changsin** [reply]
Thanks for the quick reply. That works.
For my purposes, I just needed to capture tls_failed_client events like the following.

```
    def tls_failed_client(self, data: tls.TlsData):
        print(f"tls_failed_client {data.context.server.address}")
```
